system/ee/ExpressionEngine/Service/EntryListing/EntryListing.php

```diff
@@ -345,7 +345,11 @@ protected function setupEntries()
                     break;
             }
 
-            $entries->search($search_fields, $this->search_value);
+            if (is_numeric($this->search_value) && strlen($this->search_value) < 3) {
```

Comment:
So will this fix only work for entry IDs less than 1000?

Reply:
This is strlen < 3 so less than 100 unless this is an error and we want `strlen <= 3`

Reply:
No, with 3 character we already use "normal search"

Reply:
Previously this was based on rule set up in javascript `>= 3`
Now we also want to expand search to numbers `<3`

So if the search keyword is integer, we accept any number of characters, and search by entry ID only. If it's `>=3` (so `! < 3`) we search in all fields